builder/dockerfile/dispatchers.go

```diff
@@ -729,17 +733,13 @@ func arg(b *Builder, args []string, attributes map[string]bool, original string)
 		hasDefault = false
 	}
 	// add the arg to allowed list of build-time args from this step on.
-	b.allowedBuildArgs[name] = true
+	b.allBuildArgs[name] = struct{}{}
 
-	// If there is a default value associated with this arg then add it to the
```

Comment:
It doesn't apply anymore, at least in here. There are no case where values override the others as they are stored in separate maps now.